src/compiler.c

```diff
@@ -199,19 +199,19 @@ typedef UInt4           LVar;
 
 #define INFO_FEXP(fexp)         PROF_FUNC(fexp)
 #define SET_INFO_FEXP(fexp,x)   SET_PROF_FUNC(fexp,x)
-#define NEXT_INFO(info)         PTR_BAG(info)[0]
-#define NR_INFO(info)           (*((Int*)(PTR_BAG(info)+1)))
-#define NLVAR_INFO(info)        (*((Int*)(PTR_BAG(info)+2)))
-#define NHVAR_INFO(info)        (*((Int*)(PTR_BAG(info)+3)))
-#define NTEMP_INFO(info)        (*((Int*)(PTR_BAG(info)+4)))
-#define NLOOP_INFO(info)        (*((Int*)(PTR_BAG(info)+5)))
-#define CTEMP_INFO(info)        (*((Int*)(PTR_BAG(info)+6)))
-#define TNUM_LVAR_INFO(info,i)  (*((Int*)(PTR_BAG(info)+7+(i))))
+#define NEXT_INFO(info)         PTR_BAG(info)[1]
+#define NR_INFO(info)           (*((Int*)(PTR_BAG(info)+2)))
+#define NLVAR_INFO(info)        (*((Int*)(PTR_BAG(info)+3)))
+#define NHVAR_INFO(info)        (*((Int*)(PTR_BAG(info)+4)))
+#define NTEMP_INFO(info)        (*((Int*)(PTR_BAG(info)+5)))
+#define NLOOP_INFO(info)        (*((Int*)(PTR_BAG(info)+6)))
+#define CTEMP_INFO(info)        (*((Int*)(PTR_BAG(info)+7)))
+#define TNUM_LVAR_INFO(info,i)  (*((Int*)(PTR_BAG(info)+8+(i))))
```

Comment:
Thanks. I made a similar change independently yesterday, but luckily saw your PR here before I completely finished it, and you version seems slightly nicer anyway :-).